Library/Homebrew/os/mac/xcode.rb

```diff
@@ -89,6 +89,7 @@ def without_clt?
 
       # Returns a Pathname object corresponding to Xcode.app's Developer
       # directory or nil if Xcode.app is not installed.
+      sig { returns(Pathname) }
```

Comment:
Given the comment here:

```suggestion
      # Returns a Pathname object corresponding to Xcode.app's Developer
      # directory or nil if Xcode.app is not installed.
      sig { returns(T.nilable(Pathname)) }
```

Not entirely sure why Sorbet doesn't notice that however?

Reply:
```rb
tmp =
  begin
    dir = MacOS.active_developer_dir

    if dir.empty? || dir == CLT::PKG_PATH || !File.directory?(dir)
      path = bundle_path
      path/"Contents/Developer" if path
    else
      # Use cleanpath to avoid pathological trailing slash
      Pathname.new(dir).cleanpath
    end
  end
T.assert_type! tmp, T.nilable(Pathname)
```

```
Expected a type but found T.untyped for T.assert_type! https://srb.help/7007
     107 |        T.assert_type! tmp, T.nilable(Pathname)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```